bundle/config/root.go

```diff
@@ -148,6 +151,20 @@ func (r *Root) Load(path string) error {
 		return fmt.Errorf("failed to load %s: %w", path, err)
 	}
 
+	if r.Environments != nil {
+		//TODO: add a command line notice that this is a deprecated option.
+		r.Targets = r.Environments
+	}
+
+	if r.Bundle.Environment != "" {
+		//TODO: add a command line notice that this is a deprecated option.
+		r.Bundle.Target = r.Bundle.Environment
+	}
+
+	// We do this for backward compatibility so ${bundle.environment} works correctly.
+	// TODO: remove when Environments section is not supported anymore.
+	r.Bundle.Environment = r.Bundle.Target
+
```

Comment:
Fields `bundle.Environment` / `bundle.Target` should never be set in the configuration file, they're read only.